idioms/priv-extend.md

```diff
@@ -1,31 +1,72 @@
-# Privacy for extensibility
+# `#[non_exhaustive]` & private fields for extensibility
 
 ## Description
 
-Use a private field to ensure that a struct is extensible without breaking
-stability guarantees.
+A small set of scenarios exist where a library author may want to add public
+fields to a public struct or new variants to an enum without breaking backwards
+compatibility. Rust offers two solutions:
+
+- Use `#[non_exhaustive]` on `struct`s, `enum`s, and `enum` variants.
+For extensive documentation on all the places `#[non_exhaustive]` can be used,
+see [the docs](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute).
+
+- You may add a private field to a struct to prevent it from being directly
+instantiated or matched against
```

Comment:
```suggestion
  instantiated or matched against
```

Reply:
Actually I would remove this point completely, since it is discussed as an alternative below